pipeline/backend/types/engine.go

```diff
@@ -44,6 +44,9 @@ type Engine interface {
 	// TailStep the workflow step logs.
 	TailStep(ctx context.Context, step *Step, taskUUID string) (io.ReadCloser, error)
 
+	// StopStep stop the workflow step.
+	StopStep(ctx context.Context, step *Step, taskUUID string) error
```

Comment:
we should call it what it is: DestroyStep ?

Reply:
or more CleanStep

Reply:
Wanted to be consistent with `StartStep` which is also a `create` somehow.

Reply:
Renamed it to `DestroyStep` as its also `DestroyWorkflow`